src/app/services/coin.service.ts

```diff
@@ -0,0 +1,40 @@
+import { BehaviorSubject } from 'rxjs/BehaviorSubject';
+
+import { BaseCoin } from '../coins/basecoin';
+import { SkycoinCoin } from '../coins/skycoin.coin';
+import { TestCoin } from '../coins/test.coin';
+
+export class CoinService {
+
+  currentCoin: BehaviorSubject<BaseCoin> = new BehaviorSubject<BaseCoin>(null);
+
+  private coins: BaseCoin[] = [];
+  private readonly storageKey = 'currentCoin';
+
+  constructor() {
+    this.loadCurrentCoin();
+    this.loadAvailableCoins();
+  }
+
+  changeCoin(coin: BaseCoin) {
+    this.currentCoin.next(coin);
+
+    this.saveCoin(coin);
+  }
+
+  private loadCurrentCoin() {
+    const storedCoin: string = localStorage.getItem(this.storageKey);
+    if (storedCoin) {
+      const coin: BaseCoin = JSON.parse(storedCoin);
+      this.currentCoin.next(coin);
+    }
+  }
+
+  private saveCoin(coin) {
+    localStorage.setItem(this.storageKey, JSON.stringify(coin));
```

Comment:
Only the coin ID should be saved in persistent storage, not the entire object. An example in which this would be necessary is if the URL of the node changes: if the whole object is saved and restored, updating the URL in the object of the coin would not work because the URL would be obtained from the persistent storage, no the updated object.

Reply:
Fixed.